### DISCUSSION TITLE
Raspberry Pi 32-bit

### DISCUSSION THREAD

**Discussion by acbtones** [semuconsulting/PyGPSClient#83] [General]
Hi,

Firstly, thanks for the great software!
However, I can't get it to install on Raspberry Pi 32-bit OS.
Using a RPi 3, I installed 32bit OS ("Raspberry Pi OS Full (32-bit)") and tried the installation - log attached: 32bit_pygpsclient.log
Cryptography fails to install so the PyGPSClient install fails.
Is that related to this: https://github.com/pyca/cryptography/issues/5766#issuecomment-778538695
I think that's why previous discussions in this forum have suggested Rust is required.
Using the same RPi 3, I installed 64-bit OS ("Raspberry Pi OS (64-bit)") and tried the installation - log attached: 64bit_pygpsclient.log
Successful installation.

My wish is to add PyGPSClient to an existing 32-bit RPi installation running other software that will only run on 32-bit.

Also, when starting PyGPSClient, is it possible to start the stream immediately i.e. without clicking the USB button?

Thanks,

Anthony

[32bit_pygpsclient.log](https://github.com/semuconsulting/PyGPSClient/files/12256451/32bit_pygpsclient.log)
[64bit_pygpsclient.log](https://github.com/semuconsulting/PyGPSClient/files/12256452/64bit_pygpsclient.log)

**Comment by semuadmin** [comment]
The issue is as follows:

- PyGPSClient uses the `pyspartn` library, which in turn depends on the `cryptography>=1.39` library to decrypt SPARTN messages.
- On most platforms, pip can install `cryptography` from a pre-compiled wheel (`.whl`) installation package.
- However, on some 32-bit Linux platforms (including Raspberry Pi OS 32 Lite or Full) the `cryptography` library is only available as a source code (`.tar.gz`) file, which means it has to be compiled by pip at installation time.
- This in turn requires: 
  1. Rust compiler support (*as some of cryptography is written in the Rust programming language*). This is not installed as standard on most platforms. Specifically, it requires Rust compiler (`rustc`) version 1.56 or later.
  2. A handful of additional [cryptography build dependencies](https://cryptography.io/en/latest/installation/), including `libssl-dev` and `libffi-dev`.
- At some point in the future, a wheel (`.whl`) installation package may be made available on 32-bit Linux platforms, but until then some additional installation steps are required, as described in the link below.

See https://github.com/semuconsulting/PyGPSClient/issues/84#issuecomment-1666443832

See also https://github.com/semuconsulting/pyspartn/blob/main/cryptography_installation/README.md
